src/action/list_directory.rs

```diff
@@ -251,20 +241,19 @@ impl super::Request for Request {
                         Some(path_type) => path_type,
                         None => return
                             Err(session::ParseError::malformed
-                                (Error::MissingFieldError
-                                    (String::from("path type"))))
+                                (session::MissingFieldError::new("path type")))
```

Comment:
You can simplify this a lot by using error mapping. Take a look how it is done [here][1].

[1]: https://github.com/google/rrg/blob/8516f06c5398ccb137507fc9c0a43baf85a5fb53/src/session/demand.rs#L72-L77

Reply:
Thanks, done.